src/gui/PasswordEdit.h

```diff
@@ -32,22 +44,33 @@ class PasswordEdit : public QLineEdit
     static const QColor ErrorColor;
 
     explicit PasswordEdit(QWidget* parent = nullptr);
+    virtual ~PasswordEdit();
     void enableVerifyMode(PasswordEdit* baseEdit);
     bool isPasswordVisible() const;
 
 public slots:
     void setShowPassword(bool show);
 
+protected:
+    bool event(QEvent* event) override;
+
 signals:
     void showPasswordChanged(bool show);
+    void capslockToggled(bool capslockOn);
 
 private slots:
     void updateStylesheet();
     void autocompletePassword(const QString& password);
+    void checkCapslockState();
 
 private:
     bool passwordsEqual() const;
 
+#if defined(Q_OS_UNIX) && !defined(Q_OS_MACOS)
```

Comment:
I believe you can simply use `Q_OS_LINUX`

Reply:
There is Wayland support on BSD.